### ISSUE TITLE
Very slow search

### ISSUE THREAD

**Issue by alt-romes** [haskell/error-messages#512]
It seems that the search box takes a very long time to filter results.

The script that does this filter seems excessively complex.

Do we really need such a powerful search script? Perhaps something simpler and faster would do

**Comment by david-christiansen** [comment]
I don't really have an opinion about how much complexity is excessive. From what I can tell, the box waits until the user has stopped typing for 500ms, and then filters mostly instantly, which definitely can feel a bit slow but also avoids having things get jumpy.

My main criteria for a filter script are:
1. Single JavaScript file that doesn't add any complexity to our build process and is likely to work unchanged for the next few years
2. Searches all the columns when selecting rows
3. Degrades gracefully for users without JS enabled (still a fair number of them in Haskell-land)

Have you tried turning down the delay setting in `index.html` to something like 150ms? Did that make it feel snappier for you?

If you propose a different one, I want to get feedback from @serras as well.

**Comment by alt-romes** [comment]
Aha! There's a delay, I couldn't tell as a user.

I think then 500ms is indeed a bit too slow, I can try decreasing it later today

Thanks for reviewing these. I'm making issues for these UX things as soon as I come across them, before I forget they were an issue at first

**Comment by serras** [comment]
One simple solution could be to change the filtering waiting time to something like 10ms, which would be almost instantaneous.

I actually came to a similar list of needs for the filtering part: simple to inject in our system (because we shouldn't need a JS expert), and something that works on the bare table. 95% of the "table filtering" libraries in JS require you to build a structure (usually a JSON) from which data is taken; but that means there's no easy fallback for those without JS activated.

After my search, I came to the following list of options:
1. Write our own JavaScript, which I'm sure would be short, but maybe would open a can of worms in bringing a more complex build infrastructure. Or we could do it with a single JS file, I'm not really sure.
2. Use TableFilter with some particular options, which required 0 changes to the HTML (so it works well with JS disabled).
